moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -37,27 +36,91 @@ export const MODES = [
-const CIRCLE_RADIUS = 20;
 
 const SELECTED_STYLE = {
   stroke: STROKE_COLOR,
-  strokeWidth: 2,
+  strokeWidth: 8,
   fill: FILL_COLOR,
-  fillOpacity: mapStyles.statusOpacities.selected,
+  fillOpacity: 0,
 };
 
 const HOVERED_STYLE = {
   stroke: STROKE_COLOR,
-  strokeWidth: 2,
+  strokeWidth: 8,
   fill: FILL_COLOR,
-  fillOpacity: mapStyles.statusOpacities.hovered,
+  fillOpacity: 0,
 };
 
 const DEFAULT_STYLE = {
   stroke: theme.palette.primary.main,
-  strokeWidth: 2,
-  fill: FILL_COLOR,
-  fillOpacity: mapStyles.statusOpacities.unselected,
+  strokeWidth: 4,
+  fill: theme.palette.secondary.main,
+  fillOpacity: 1,
+};
+
+/**
+ * Interpolate a feature width based on the zoom level of map
+ * Adapted from Mapbox GL JS linear interpolation using a formula linked below
+ * https://github.com/mapbox/mapbox-gl-js/blob/d66ff288e7ab2e917e9e676bee942dd6a46171e7/src/style-spec/expression/definitions/interpolate.js
+ * https://matthew-brett.github.io/teaching/linear_interpolation.html
+ * @param {number} currentZoom - Current zoom level from the map
+ * @param {number} minZoom - Minimum zoom level from the current bracket
+ * @param {number} maxZoom - Maximum zoom level from the current bracket
+ * @param {number} minPixelWidth - Minimum pixel width from the current bracket
+ * @param {number} maxPixelWidth - Maximum pixel width from the current bracket
+ * @return {number} Interpolated pixel width
+ */
+function linearInterpolation(
+  currentZoom,
+  minZoom,
+  maxZoom,
+  minPixelWidth,
+  maxPixelWidth
+) {
+  return (
+    ((currentZoom - minZoom) * (maxPixelWidth - minPixelWidth)) /
+      (maxZoom - minZoom) +
+    minPixelWidth
+  );
+}
+
+/**
+ * Calculate the circle radius using the circle radius steps in mapStyles and the map zoom level
+ * @param {number} currentZoom - Current zoom level from the map
+ * @return {number} Circle radius in pixels
+ */
+const getCircleRadiusByZoom = currentZoom => {
+  const { stops } = mapStyles.circleRadiusStops;
+  const [bottomZoom, bottomPixelWidth] = stops[0];
+  const [topZoom, topPixelWidth] = stops[stops.length - 1];
+
+  // Loop through the [zoom, radius in pixel] nested arrays in mapStyles.circleRadiusStops
+  // to find which two elements the current zoom level falls between
+  for (let i = 0; i < stops.length - 1; i++) {
+    const [minZoom, minPixelWidth] = stops[i];
+    const [maxZoom, maxPixelWidth] = stops[i + 1];
+
+    // If current zoom is less than zoom in the first element
+    if (currentZoom < bottomZoom) {
+      return bottomPixelWidth;
+    }
+
+    // If current zoom is greater than zoom in the last element
+    if (currentZoom >= topZoom) {
+      return topPixelWidth;
+    }
+
+    // If the current zoom falls somewhere between
+    if (currentZoom >= minZoom && currentZoom < maxZoom) {
+      return linearInterpolation(
+        currentZoom,
+        minZoom,
+        maxZoom,
+        minPixelWidth,
+        maxPixelWidth
+      );
+    }
+  }
```

Comment:
Ah, yes the for loop. I personally don't mind, but mateo might ask you why not use `map` or `forEach` ? 🤷‍♂️

Reply:
Just seeing this, yep, the reason that I used a `for` loop is that the loop is used to search where a value falls in some ranges. I wanted to be able to break out of the loop once the match is found and prevent unnecessary iterations since this function fires rapidly when a user is zooming in and out